### PR TITLE
snapstate,store: store if a snap is a paid snap in the sideinfo

### PR DESCRIPTION
**Pull request by mvo5**
Keep the data about payed snaps in the state just like we do with
private snaps.

### CODE REVIEW

File: snap/info.go [hunk 1]
```diff
@@ -135,6 +135,7 @@ type SideInfo struct {
 	EditedSummary     string   `yaml:"summary,omitempty" json:"summary,omitempty"`
 	EditedDescription string   `yaml:"description,omitempty" json:"description,omitempty"`
 	Private           bool     `yaml:"private,omitempty" json:"private,omitempty"`
+	Payed             bool     `yaml:"payed,omitempty" json:"payed,omitempty"`
```
Comment (bboozzoo) [inline snap/info.go]: I'm not native English speaker, but I think you wanted 'Paid' here

File: overlord/snapstate/backend_test.go [hunk 2]
```diff
@@ -146,6 +140,18 @@ func (f *fakeStore) SnapInfo(spec store.SnapSpec, user *auth.UserState) (*snap.I
 		Confinement: confinement,
 		Type:        typ,
 	}
+	switch spec.Channel {
+	case "channel-for-devmode":
+		info.Confinement = snap.DevModeConfinement
+	case "channel-for-classic":
+		info.Confinement = snap.ClassicConfinement
+	case "channel-for-payed":
+		info.Prices = map[string]float64{"USD": 0.77}
```
Comment (bboozzoo) [inline overlord/snapstate/backend_test.go]: Hopefully no real financial operations are done with money expressed as float :)

**Comment by chipaca** [comment]
can confirm it should be `paid` everywhere

File: overlord/snapstate/backend_test.go (hunk 2 above)
Comment (chipaca) [reply overlord/snapstate/backend_test.go]: I too was surprised to find out that yes, some of these things are done with floats. Not AFAIK in aggregate though.

File: snap/info.go [hunk 3]
```diff
@@ -135,6 +135,7 @@ type SideInfo struct {
 	EditedSummary     string   `yaml:"summary,omitempty" json:"summary,omitempty"`
 	EditedDescription string   `yaml:"description,omitempty" json:"description,omitempty"`
 	Private           bool     `yaml:"private,omitempty" json:"private,omitempty"`
+	Paid              bool     `yaml:"paid,omitempty" json:"paid,omitempty"`
```
Comment (zyga) [inline snap/info.go]: I wonder if this means `this snap is for sale` or `this snap was purchased`; Perhaps we could have a comment here.

File: store/store_test.go [hunk 4]
```diff
@@ -4648,6 +4660,7 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreDecorateOrdersTokenExpired(c *C) {
 	helloWorld := &snap.Info{}
 	helloWorld.SnapID = helloWorldSnapID
 	helloWorld.Prices = map[string]float64{"USD": 1.23}
+	helloWorld.Paid = true
```
Comment (zyga) [inline store/store_test.go]: I must say that in all of the diff snippets above I was surprised we didn't just say:
```
helloWorld := &snap.Info{
   SnapID: ...
}
```

File: snap/info.go (hunk 3 above)
Comment (pedronis) [reply snap/info.go]: it means both/either actually depending on context

**Pull request merged**